### PR TITLE
Sign up functionality with Firebase

### PR DESCRIPTION
**Pull request by adshayanB**
Added signup functionality with Firebase

https://github.com/adshayanB/ECE452/assets/41243143/8390b8b0-b95b-450c-b9f2-02041ee1abc4

Also added onclick between screens using navigation routes

https://github.com/adshayanB/ECE452/assets/41243143/dfe828ed-ee5d-4216-b713-175debf220f7

### CODE REVIEW

File: app/src/main/java/com/example/farmeraid/data/UserRepository.kt [hunk 1]
```diff
@@ -23,4 +27,21 @@ class UserRepository {
...
+
+    suspend fun signup(userName: String, password: String) : SignInModel.AuthResponse {
+        val firebaseAuth: FirebaseAuth = FirebaseAuth.getInstance()
+        if (userName.isNotEmpty() && password.isNotEmpty()) {
+            return try {
+                val res = firebaseAuth.createUserWithEmailAndPassword(userName, password).await()
+                SignInModel.AuthResponse.Success
+            } catch (e: Exception) {
+                return SignInModel.AuthResponse.Error(
+                    e.message ?: "Error logging in. Please try again later."
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/data/UserRepository.kt:39]: Change this to "Error signing up. Please try again later."

File: app/src/main/java/com/example/farmeraid/uicomponents/BottomNavigationBar.kt [hunk 2]
```diff
@@ -46,6 +46,7 @@ fun BottomNavigationBar(appNavigator: AppNavigator) {
     )
     val notShownScreens: List<String> = listOf(
         NavRoute.SignIn.route,
+        NavRoute.SignUp.route
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/uicomponents/BottomNavigationBar.kt:49]: Nice catch adding this in here. Forgot to mention that you need to do this.

File: app/src/main/java/com/example/farmeraid/sign_up/SignUpViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    fun signup(userName: String, password: String) = viewModelScope.launch {
+        buttonUiState.value = buttonUiState.value.copy(isLoading = true)
+        val result: SignInModel.AuthResponse = userRepository.signup(userName, password)
+        buttonUiState.value = buttonUiState.value.copy(isLoading = false)
+
+        when(result) {
+            is SignInModel.AuthResponse.Success -> {
+                Log.d("MESSAGE", "LOGGED IN")
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/sign_up/SignUpViewModel.kt:64]: Change this to "SIGNED UP"
and show a snackbar saying "Account created successfully."

File: app/src/main/java/com/example/farmeraid/sign_up/SignUpViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        }
+    }
+
+    fun signup(userName: String, password: String) = viewModelScope.launch {
+        buttonUiState.value = buttonUiState.value.copy(isLoading = true)
+        val result: SignInModel.AuthResponse = userRepository.signup(userName, password)
+        buttonUiState.value = buttonUiState.value.copy(isLoading = false)
+
+        when(result) {
+            is SignInModel.AuthResponse.Success -> {
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/sign_up/SignUpViewModel.kt:63]: Change this to SignUpModel.AuthResponse

File: app/src/main/java/com/example/farmeraid/data/UserRepository.kt [hunk 5]
```diff
@@ -23,4 +27,21 @@ class UserRepository {
             return SignInModel.AuthResponse.Error("Fields cannot be empty")
         }
     }
+
+    suspend fun signup(userName: String, password: String) : SignInModel.AuthResponse {
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/data/UserRepository.kt:31]: Change the return to SignUpModel.AuthResponse

**Review by RishanR — CHANGES_REQUESTED** [review]
Overall looking good! Just a few minor changes.

File: app/src/main/java/com/example/farmeraid/uicomponents/BottomNavigationBar.kt (hunk 2 above)
Comment (adshayanB) [reply app/src/main/java/com/example/farmeraid/uicomponents/BottomNavigationBar.kt:49]: Ye no worries, don't want nav bar to show on signup XD

**Review by RishanR — APPROVED** [review]
LGTM! Great job!

**Pull request merged**